### ISSUE TITLE
PHPCBF fails to fix file with empty statement at start on control structure

### ISSUE THREAD

**Issue by kkmuffme** [squizlabs/PHP_CodeSniffer#2810]
phpcf gets stuck in an inifinite loop if the code contains "{;"  (which can happen by accident)
It' will do loads of passes where it encounters 0/ violations remaining but then again there are 1, 2, many violations so it never ends.

Sample code:

```php
if ( 2 === 2 ) {;
    echo "hello";
}
```

Sample output:

```
Registering sniffs in the abc standard... DONE (248 sniffs registered)
Creating file list... DONE (1 files in queue)
Changing into directory /some/path/
Processing test.php [PHP => 25 tokens in 5 lines]... DONE in 103ms (2 fixable violations)
        => Fixing file: 1/2 violations remaining [made 34 passes]... ^C
```

===

Suggested fix:
1) solve that it doesn't end up stuck
2) additionally if a line ends with {; the ; can be removed?

**Comment by gsherwood** [comment]
This is replicated using the included Squiz standard, and is a conflict between `Squiz.ControlStructures.ControlSignature` and `Squiz.WhiteSpace.SemicolonSpacing`.

**Comment by gsherwood** [comment]
I've committed a fix for this. Given this is an empty statement, the `Squiz.Whitespace.SemicolonSpacing `sniff will now ignore it. The code will be fixed like this:
```php
if ( 2 === 2 ) {
    ;
    echo "hello";
}
```
The `Generic.CodeAnalysis.EmptyPHPStatement` can be used to detect and fix the empty statement, although I had to commit a fix to get it working for this specific case. That will also be in 3.5.5.

Thanks for the report.
